src/main/java/net/snowflake/client/core/json/StringConverter.java

```diff
@@ -0,0 +1,134 @@
+package net.snowflake.client.core.json;
+
+import java.sql.Date;
+import java.sql.Types;
+import java.util.TimeZone;
+import net.snowflake.client.core.ResultUtil;
+import net.snowflake.client.core.SFException;
+import net.snowflake.client.jdbc.ErrorCode;
+import net.snowflake.client.jdbc.SnowflakeUtil;
+import net.snowflake.client.log.ArgSupplier;
+import net.snowflake.client.log.SFLogger;
+import net.snowflake.client.log.SFLoggerFactory;
+import net.snowflake.common.core.*;
+
+public class StringConverter {
+  private static final SFLogger logger = SFLoggerFactory.getLogger(StringConverter.class);
+  private final TimeZone sessionTimeZone;
+  private final SFBinaryFormat binaryFormatter;
+  private final SnowflakeDateTimeFormat dateFormatter;
+  private final SnowflakeDateTimeFormat timeFormatter;
+  private final SnowflakeDateTimeFormat timestampNTZFormatter;
+  private final SnowflakeDateTimeFormat timestampLTZFormatter;
+  private final SnowflakeDateTimeFormat timestampTZFormatter;
+  private final long resultVersion;
+  private final Converters converters;
+
+  public StringConverter(
+      TimeZone sessionTimeZone,
+      SFBinaryFormat binaryFormatter,
+      SnowflakeDateTimeFormat dateFormatter,
+      SnowflakeDateTimeFormat timeFormatter,
+      SnowflakeDateTimeFormat timestampNTZFormatter,
+      SnowflakeDateTimeFormat timestampLTZFormatter,
+      SnowflakeDateTimeFormat timestampTZFormatter,
+      long resultVersion,
+      Converters converters) {
+    this.sessionTimeZone = sessionTimeZone;
+    this.binaryFormatter = binaryFormatter;
+    this.dateFormatter = dateFormatter;
+    this.timeFormatter = timeFormatter;
+    this.timestampNTZFormatter = timestampNTZFormatter;
+    this.timestampLTZFormatter = timestampLTZFormatter;
+    this.timestampTZFormatter = timestampTZFormatter;
+    this.resultVersion = resultVersion;
+    this.converters = converters;
+  }
+
+  public String getString(Object obj, int columnType, int columnSubType, int scale)
+      throws SFException {
+    if (obj == null) {
+      return null;
+    }
+
+    switch (columnType) {
+      case Types.BOOLEAN:
+        return ResultUtil.getBooleanAsString(ResultUtil.getBoolean(obj.toString()));
+
+      case Types.TIMESTAMP:
+      case SnowflakeUtil.EXTRA_TYPES_TIMESTAMP_LTZ:
+      case SnowflakeUtil.EXTRA_TYPES_TIMESTAMP_TZ:
+        SFTimestamp sfTS =
```

Comment:
What do you think about extracting all under case to method and use as one line here

Reply:
Yes, this is a good idea.